plugins/available/basher.plugin.bash

```diff
@@ -0,0 +1,14 @@
+cite about-plugin
+about-plugin 'initializes basher, the shell package manager'
+
+# https://github.com/basherpm/basher
+
+_command_exists basher \
+	|| [[ -x "$HOME/.basher/bin/basher" ]] \
+	|| return 0
+
+if ! _command_exists basher && [[ -x "$HOME/.basher/bin/basher" ]]; then
```

Comment:
You check the same conditions twice, maybe we can simplify this by putting the checks in an `if` statement

Reply:
I'd rather keep the pattern of checking conditions and bailing at the top before any logic is executed. I thank @davidpfarrell for setting up this pattern when they reviewed the `goenv` plugin.

Reply:
@NoahGorny I've update the code as requested, but I think we should discuss this pattern in the abstract. I think patterns are great for projects that span lots of contributors, and I don't like how many plugins currently do stuff before checking if they have the tools necessary to complete the rest of the script.

Having the first line check for short circuits is a *huge* improvement to me.